compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -275,10 +274,9 @@ pub unsafe fn create_module<'ll>(
         }
     }
 
-    if let Some(BranchProtection { bti, pac_ret }) = sess.opts.unstable_opts.branch_protection {
-        if sess.target.arch != "aarch64" {
-            sess.emit_err(BranchProtectionRequiresAArch64);
-        } else {
+    // AArch64-only options (checked in rustc_session).
+    if sess.target.arch == "aarch64" {
+        if let Some(BranchProtection { bti, pac_ret }) = sess.opts.unstable_opts.branch_protection {
```

Comment:
Rather than making this a no-op, we should probably `bug!` this, so that any issues with checking in `rustc_session` don’t get silently ignored.

Reply:
Good point! Done in [73d374f](https://github.com/rust-lang/rust/pull/105421/commits/73d374f3e7d5fe9635a23937cdc2f7e1d1c29224).